src/Symfony/Component/Routing/Generator/ConfigurableQueryEncodingTypeInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Generator;
+
+/**
+ * ConfigurableQueryEncodingTypeInterface must be implemented by URL generators
+ * that can be configured to build query strings by replacing spaces with their
+ * percent-encoded value (%20) according to RFC3986 or by replacing them with a
+ * plus sign (+) according to RFC1738.
+ *
+ * @author Jonathan Cox <jonathan.cox@fostermade.co>
+ */
+interface ConfigurableQueryEncodingTypeInterface
```

Comment:
I'm not sure adding this interface is a good idea. What about adding a constructor argument instead?

Reply:
My understanding was that adding an argument would be a BC break for anyone who may have extended the `UrlGenerator`, so I mirrored the way configurable requirements were implemented. I see that `defaultEncoding` was added in `4.1` (it wasn’t there when I wrote this in December) — so I’ll happily change it if my understanding of the BC implications is wrong.

Reply:
adding argument *on the constructor* is fone

Reply:
And after more than a moment’s thought, the reason behind that is perfectly clear. Thanks!